File: shared/src/main/scala/stasis/shared/api/requests/UpdateDatasetDefinition.scala
package stasis.shared.api.requests

import java.time.Instant

import stasis.shared.model.datasets.DatasetDefinition

final case class UpdateDatasetDefinition(
  info: String,
  redundantCopies: Int,
  existingVersions: DatasetDefinition.Retention,
  removedVersions: DatasetDefinition.Retention
)

object UpdateDatasetDefinition {
  implicit class RequestToUpdatedDefinition(request: UpdateDatasetDefinition) {
    def toUpdatedDefinition(definition: DatasetDefinition): DatasetDefinition =
      definition.copy(
        info = request.info,
        redundantCopies = request.redundantCopies,
        existingVersions = request.existingVersions,
        removedVersions = request.removedVersions,
        updated = Instant.now()
      )
  }
}

File: client-ios/lib/Sources/StasisClientLib/Model/Server/Api/Requests/UpdateDatasetDefinition.swift
import Foundation

public struct UpdateDatasetDefinition: Sendable, Equatable, Hashable, Codable {
    public let info: String
    public let redundantCopies: Int
    public let existingVersions: DatasetDefinition.Retention
    public let removedVersions: DatasetDefinition.Retention

    public init(
        info: String,
        redundantCopies: Int,
        existingVersions: DatasetDefinition.Retention,
        removedVersions: DatasetDefinition.Retention
    ) {
        self.info = info
        self.redundantCopies = redundantCopies
        self.existingVersions = existingVersions
        self.removedVersions = removedVersions
    }
}
